_posts/2017-01-16-Jak-psat-php-Uloziste-kodu+databaze+frameworky.md

```diff
@@ -0,0 +1,46 @@
+
+Mezi dalšími chci zmínit [CodePlex](https://www.codeplex.com/) Mezi služby, které nechi doporučovat patří [SourceForge](https://sourceforge.net/), který měl veliké problémy s přibalováním nevyžádaných probramů do všech projektů. Používal se i [Google Code](https://code.google.com/archive/), který ale v roce 2016 ukončil svou činnost a je dostupný jen ve formě archívu. 
+
+# Databáze
+Databáze jsou nedílnou součástí PHP i webových aplikací obecně, přesto je vezmu velmi zkrátka - je to disciplína hodná vlastního blogu (šance založit jakpsatdb.cz!). 
+
+Databáze slouží k ukládání dat. Data jdou řešit i jinak, například ukládat do textových souborů či posílat někam mimo aplikaci. Výhodou databází je ale přístupová rychlost a snadná udržitelnost a zálohovatelnost. 
+
+Databáze se obecně dělí na strukturované (relační) a nestukturované. Relační databáze mají pevně danou strukturu tabulek a sloupců v nich. Asi nejpoužívanější je [MySQL](https://www.mysql.com/) od firmy Sun. Další známé systémy jsou [MariaDB](https://mariadb.org/) jako opensource klon MySQL, oblíbený [PostgresQL](https://www.postgresql.org/), [Firebird](http://www.firebirdsql.org/) používaný hlavně v komerční sféře a jednoduchý [SQLite](https://sqlite.org/) vhodný pro malé projekty. 
+
+Do nestukturovaných databází (známých i jako noSQL) se většinou informaci ukládají ve tvaru `klíč:hodnota`. Používají se zejména tam, kde by byla nejasná či příliš složitá struktua relační databáze či při požadavcích na velkou rychlost a objem v jednoduché struktuře, jako je tomu například u dat generovaných z Internet of Things (internetu věcí). Mezi používané NoSQL systémy se řadí [MongoDB](https://www.mongodb.com/) nebo [Redis](https://redis.io/). 
+
+Z pohledu ovládání databáze můžete používat funkce přímo z PHP, jako třeba [mysqli](https://secure.php.net/manual/en/book.mysqli.php) nebo [PDO](https://secure.php.net/manual/en/book.pdo.php). Často se také používá nějaká externí knihovna. Knihovny můžou pracovat s jazykem SQL napřímo, jako kupříkladu [Dibi](https://www.dibiphp.com/), nebo pomocí objektově relačního mapování (ORM), jako třeba [Doctrine](http://www.doctrine-project.org/). 
+
+Pro ovládání databáze uživatelem se nejvíce používá [phpMyAdmin](https://www.phpmyadmin.net/). Za sebe můžu doporučit mnohem lehčí [adminer](https://www.adminer.org/cs/) od [Jakuba Vrány](https://www.vrana.cz/). 
+
+# Frameworky
+Ve vývoji webových aplikací je spousta požadavků na funkčnosti, které se prolínají většinou aplikací. Tyto funkce jsou od zkušenějších vývojářů už bezpočetněkrát popsané, vymyšlené a napsané. Neustále probíhají veliké diskuze, jak by se tyto standardní problémy měly nejlépe řešit. A frameworky jsou sady kódu, které by měly tyto běžné problémy implementovat pomocí nejlepšího dostupného řešení. 
```

Comment:
Tady bych asi neporozuměl tomu, jak se liší PHP a framework. Proč ho mám používat? Co se mám učit?...